hashing/ripemd_160.cpp

```diff
@@ -0,0 +1,457 @@
+#include <cstdint>   /// For data types such as int32_t, uint32_t, etc
+#include <iomanip>   /// For functions like setw, setfill
+#include <iostream>  /// For managing io
+#include <sstream>   /// For bytes to hex string
+#include <string>    /// For string data
+#include <thread>    /// To parallelize for efficiency
+#include <vector>    /// For dynamic arrays
+
+/**
+ * @namespace hashing
+ * @brief Hashing algorithms
+ */
+namespace hashing {
+/**
+ * @class RIPEMD-160
+ * @brief Implementation of relevant functions to perform RIPEMD160 hashing.
+ * This class expects a string to hash.
+ */
+class RIPEMD160 {
+ private:
+    /**
+     * @brief converts string data to vector of uint32_t (4 byte words)
+     * @details converts the string to 4 byte words in little endian format
+     * after adding padding and message length as specified in the algorithm
+     * @param data data to be converted
+     * @return vector of 4 byte words in little endian format
+     */
+    std::vector<uint32_t> string_to_word_data(std::string &data) {
+        std::vector<uint32_t> word_data;
+
+        // convert string data to 4 byte words in little endian format
+        for (uint64_t i = 0; i < (data.size() / 4) * 4; i += 4) {
+            word_data.push_back(static_cast<uint32_t>(data[i]) |
+                                (static_cast<uint32_t>(data[i + 1]) << 8) |
+                                (static_cast<uint32_t>(data[i + 2]) << 16) |
+                                (static_cast<uint32_t>(data[i + 3]) << 24));
+        }
+
+        // add padding to data
+        int extra = data.size() % 4;
+
+        if (extra == 0) {
+            word_data.push_back(0x00000080);
+        }
+
+        else if (extra == 1) {
+            word_data.push_back(static_cast<uint32_t>(data[data.size() - 1]) |
+                                0x00008000);
+        }
+
+        else if (extra == 2) {
+            word_data.push_back(
+                static_cast<uint32_t>(data[data.size() - 2]) |
+                (static_cast<uint32_t>(data[data.size() - 1]) << 8) |
+                0x00800000);
+        }
+
+        else if (extra == 3) {
+            word_data.push_back(
+                static_cast<uint32_t>(data[data.size() - 3]) |
+                (static_cast<uint32_t>(data[data.size() - 2]) << 8) |
+                (static_cast<uint32_t>(data[data.size() - 1]) << 16) |
+                0x80000000);
+        }
+
+        while (word_data.size() % 16 != 14) {
+            word_data.push_back(0x00000000);
+        }
+
+        // add message length
+        word_data.push_back((static_cast<uint64_t>(data.size()) << 3) &
+                            0xffffffff);
+        word_data.push_back((static_cast<uint64_t>(data.size()) >> 29) &
+                            0xffffffff);
+
+        return word_data;
+    }
+
+    /**
+     * @brief implements f(j,x,y,z)
+     * @param j round number j / 16
+     * @param x value x
+     * @param y value y
+     * @param z value z
```

Comment:
```suggestion
     * @param x,y,z the coordinate space
```

please be more detailed about the parameters, and what they stand for
its really vague right now.